src/unix/atomic-ops.h

```diff
@@ -18,6 +18,11 @@
 
 #include "internal.h"  /* UV_UNUSED */
 
+#if defined(__SUNPRO_C)
```

Comment:
Because this is a header file. I believe you will need to add || defined(__SUNPRO_CC) since it can also be compiled via a C++ compiler